src/validate.js

```diff
@@ -0,0 +1,36 @@
+import { isObject, toErrorList, mergeObjects } from "./utils";
+
+
+function createErrorHandler(formData) {
+  const handler = {
+    __errors: [],
+    addError(message) {
+      this.__errors = [...this.__errors, message];
```

Comment:
I won't go the microbenchmark route, but I have a hunch the impact on perfs would be rather small here. Anyway I can switch back to `push` anyway if that makes you more comfortable with the change.